src/model.cpp

```diff
@@ -609,7 +609,11 @@ std::vector<bool> const &Model::getStateIsNonNegative() const {
 }
 
 void Model::setStateIsNonNegative(std::vector<bool> const &nonNegative) {
-    if (nx_solver != nx_rdata) {
+    auto any_state_non_negative = std::any_of(state_is_non_negative_.begin(),
```

Comment:
I am not sure what's implemented here, shouldn't this be applied to `nonNegative`? Not implementing this was primarily a result of lazyness. If we specify what kind of indices are passed here (solver or rdata x), we could also support setting non_negative states, just need to map between the two.

Reply:
True. Thanks.

I'll open a separate issue for that.

Reply:
> If we specify what kind of indices are passed here (solver or rdata x), we could also support setting non_negative states, just need to map between the two.

With x_solver indices we wouldn't have to check whether it actually is a solver state. However, I'd guess that most users wouldn't want to care about x_solver at all, and therefore, x_rdata would be more appropriate. Or we just do it ID-based. That's probably the most convenient. What would be your preference?

Reply:
> > If we specify what kind of indices are passed here (solver or rdata x), we could also support setting non_negative states, just need to map between the two.
> 
> With x_solver indices we wouldn't have to check whether it actually is a solver state. However, I'd guess that most users wouldn't want to care about x_solver at all, and therefore, x_rdata would be more appropriate. Or we just do it ID-based. That's probably the most convenient. What would be your preference?

I agree, I would expect that users want to pass x_rdata. For id based specification we should probably add another function as changing this function would be API breaking.